ddht/v5_1/network.py

```diff
@@ -403,21 +404,67 @@ async def find_nodes(
         # Validate that all responses are indeed at one of the
         # specified distances.
         for response in responses:
-            for enr in response.message.enrs:
-                if enr.node_id == node_id:
-                    if 0 not in distances:
-                        raise ValidationError(
-                            f"Invalid response: distance=0  expected={distances}"
-                        )
-                else:
-                    distance = compute_log_distance(enr.node_id, node_id)
-                    if distance not in distances:
-                        raise ValidationError(
-                            f"Invalid response: distance={distance}  expected={distances}"
-                        )
+            validate_found_nodes_distances(response, node_id, distances)
 
         return tuple(enr for response in responses for enr in response.message.enrs)
 
+    @asynccontextmanager
+    async def stream_find_nodes(
+        self,
+        node_id: NodeID,
+        endpoint: Endpoint,
+        distances: Collection[int],
+        *,
+        request_id: Optional[bytes] = None,
+    ) -> AsyncIterator[trio.abc.ReceiveChannel[InboundMessage[FoundNodesMessage]]]:
```

Comment:
This should be streaming the individual ENR records extracted from the responses you are streaming from the `ClientAPI.stream_find_nodes`